File: 0199.binary_tree_right_side_view/solution.py
from typing import List

import utils


class Solution:
    def right_side_view(self, root: utils.TreeNode) -> List[int]:
        """ 先层序遍历一遍二叉树，每层获取最右边值"""
        if not root:
            return []
        stage = [(0, root)]
        level = []
        while stage:
            count, node = stage.pop(0)
            if len(level) <= count:
                level.append([node.val])
            else:
                level[count].append(node.val)
            if node.right:
                stage.append((count + 1, node.right))
            if node.left:
                stage.append((count + 1, node.left))

        ans = [i[0] for i in level]
        return ans

File: 0199.binary_tree_right_side_view/solution.go
package leetcode0199

import "github.com/Ackerr/Algorithms/utils"

func rightSideView(root *utils.TreeNode) []int {
	res := make([]int, 0)
	if root == nil {
		return res
	}
	nodes := []*utils.TreeNode{root}
	for len(nodes) > 0{
		n := len(nodes)
		for i:=0;i<n;i++ {
			p := nodes[i]
			if p.Right != nil {
				nodes = append(nodes, p.Right)
			}
			if p.Left != nil {
				nodes = append(nodes, p.Left)
			}
			if i == 0{
				res = append(res, p.Val)
			}
		}
		nodes = nodes[n:]

	}
	return res
}
